app/src/main/java/com/example/healthinspector/Fragments/ScanFlow/ProductDetailsFragment.java

```diff
@@ -66,6 +82,26 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         ArrayAdapter warningsAdapter = new ArrayAdapter<String>(requireContext(),
                android.R.layout.simple_list_item_1,scannedProduct.getNutrientLevels());
         binding.warningsListView.setAdapter(warningsAdapter);
+
+        ArrayList<String> additivesInProduct;
+        try {
+            allAdditives = CachedLists.getInstance(requireContext()).getAdditives();
+            if(scannedProduct.getAdditives().size() >= 1){
+                additivesInProduct = new ArrayList<>();
+                for(int i = 0; i < scannedProduct.getAdditives().size(); i++){
+                    if(allAdditives.containsKey(scannedProduct.getAdditives().get(i))){
+                        additivesInProduct.add(allAdditives.get(scannedProduct.getAdditives().get(i)));
+                    }
+                }
```

Comment:
Couple questions here:
- are we doing this because the API only returns the ingredient code like `en:e330` and we need to find the actual ingredient name like E330 - Citric acid?
- since `getAdditives` returns a hashmap, could we not just use the key to access the value instead of iterating through the entire hashmap?

Reply:
Yes, the API doesn't return the actual name of the ingredient like you mentioned above it only gives a key to the additives included in the product (like so: "additives_original_tags": [
            "en:e503i",
            "en:e500ii",
            "en:e322i"
        ] ).
The getAdditives method returns ArrayList<String>

Reply:
I thought's it's this function:
https://github.com/OmarBafagih/Health-Inspect/blob/main/app/src/main/java/com/example/healthinspector/CachedLists.java#L33

and it returns a HashMap<String, String>

Reply:
The function I'm using here is actually from the scannedProduct model class where it returns the stored list of additives contained in the product (from the API), I should rename one of them so that there's less confusion

Reply:
From what I understand,
`allAdditives = CachedLists.getInstance(requireContext()).getAdditives();`
uses the function I linked above
`if(scannedProduct.getAdditives().size() >= 1){`
uses the function from scannedProduct model

Reply:
Yes, that's exactly it

Reply:
so what I'm saying is that we could do something like:
```
key = scannedProduct(i);
allAdditives.getValue(key)
```